optuna/pruners/successive_halving.py

```diff
@@ -72,10 +72,18 @@ class SuccessiveHalvingPruner(BasePruner):
             A parameter for specifying the minimum early-stopping rate
             (in the `paper <http://arxiv.org/abs/1810.05934>`_ this parameter is
             referred to as :math:`s`).
+        rung_key_prefix:
```

Comment:
IMO, it's better to remove the `rung_key_prefix` option if possible because this option is beneficial only for a very limited purpose (i.e., implementing `HyperbandPruner`) and is noisy to library users since it's not relevant to them.

This is a just idea though, how about filtering trials on `HyperbandPruner` side by introducing a new study class as follows?
```python
# file: optuna/pruners/hyperband.py

class HyperbandPruner(BasePruner):
  ...

  def prune(self, study, trial):
      i = self.get_bracket_id(study.study_name, trial.number)
      bracket_study = BracketStudy(study, i)
      return self._pruners[i].prune(bracket_study, trial)
  

class BracketStudy(Study):
  def __init__(self, inner_study, bracket_id):
    self._inner_study = inner_study
    self._bracket_id = bracket_id

  def trials(self):
    return [t for t in self._inner_study.trials if get_bracket_id(self._inner_study.name, t.number) == self._bracket_id]
```

Reply:
OK, although the CIs fail, could you take a look on the current design?

Reply:
Thanks! I'll review this PR again tomorrow.